src/com/teamcautionrobotics/robot2018/MissionSelector.java

```diff
@@ -0,0 +1,191 @@
+                }
+            }
+        } else if (autoObjective == AutoObjective.SCALE) {
+            if (startingPosition == StartingPosition.CENTER) {
+                if (scaleSide == PlateSide.LEFT) {
+                    return centerMissionLeftScale;
+                } else if (scaleSide == PlateSide.RIGHT) {
+                    return centerMissionRightScale;
+                }
+            } else if (startingPosition == StartingPosition.LEFT) {
+                if (scaleSide == PlateSide.LEFT) {
+                    return leftMissionLeftScale;
+                } else if (scaleSide == PlateSide.RIGHT) {
+                    return driveForwardMission;
+                }
+            } else if (startingPosition == StartingPosition.RIGHT) {
+                if (scaleSide == PlateSide.LEFT) {
+                    return driveForwardMission;
+                } else if (scaleSide == PlateSide.RIGHT) {
+                    return rightMissionRightScale;
+                }
+            }
+        } else if (autoObjective == AutoObjective.SWITCH_OR_SCALE) {
+            if (startingPosition == StartingPosition.CENTER) {
+                if (switchSide == PlateSide.LEFT) {
+                    return centerMissionLeftSwitch;
+                } else if (switchSide == PlateSide.RIGHT) {
+                    return centerMissionRightSwitch;
+                }
+            } else if (startingPosition == StartingPosition.LEFT) {
+                if (switchSide == PlateSide.LEFT) {
+                    return leftMissionLeftSwitch;
+                } else if (switchSide == PlateSide.RIGHT) {
+                    if (scaleSide == PlateSide.LEFT) {
+                        return leftMissionLeftScale;
+                    } else if (scaleSide == PlateSide.RIGHT) {
+                        return driveForwardMission;
+                    }
+                }
+            } else if (startingPosition == StartingPosition.RIGHT) {
+                if (switchSide == PlateSide.LEFT) {
+                    if (scaleSide == PlateSide.LEFT) {
+                        return driveForwardMission;
+                    } else if (scaleSide == PlateSide.RIGHT) {
+                        return rightMissionRightScale;
+                    }
+                } else if (switchSide == PlateSide.RIGHT) {
+                    return rightMissionRightSwitch;
+                }
+            }
+        } else if (autoObjective == AutoObjective.DO_NOTHING) {
+            return DO_NOTHING_MISSION;
+        }
+        return new Mission("Mission selection error", new Command() {
+            @Override
+            public boolean run() {
+                System.err.format(
+                        "Unable to select mission. Switch side: %s, scale side: %s, "
+                                + "starting position: %s, auto objective: %s%n",
+                        switchSide.toString(), scaleSide.toString(), startingPosition.toString(),
```

Comment:
We should probably throw an exception here instead of the error mission, but this does not need to be changed to be merged.